### DISCUSSION TITLE
What is the recommended way to get a proper URL to an object in an S3 bucket?

### DISCUSSION THREAD

**Discussion by kevmodrome** [pocketbase/pocketbase#26] [Q&A] [answered]
Title 👆

Uploading works fine, but how do you get the objects out? Any ideas?

**Comment by ganigeorgiev** [comment, marked as answer]
You don't have to interact with the S3 endpoint directly. Requesting the files api url will make the S3 call for you (using the S3 API) and will return your object:
```
http://localhost:8090/api/files/COLLECTION_ID_OR_NAME/RECORD_ID/FILENAME
```

This may not be suitable for all S3 providers because some of them apply rate limits and the usual recommendation is to use a CDN (your objects will need to be publicly accessible).
In that case you would have to append the s3 object key (aka. `/COLLECTION_ID/RECORD_ID/FILENAME`) to the CDN address (notice that only the collection id could be used here).

The main drawback of this approach is that you'll loose the thumb generation feature which is available only when you request the files api with the `?thumbs=WxH` query parameter.

One possible solution to that could be to add a CDN field in the S3 config and make the files api to redirect to it.

**Reply by ganigeorgiev** [reply]
I've created an issue to add the CDN field in the S3 configuration - #28

**Reply by kevmodrome** [reply]
Thanks! Looks like I was accidentally using the collectionID as the recordId 🤦‍♂️
